specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/library.json

```diff
@@ -0,0 +1,210 @@
+          "description": "Error code.",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message.",
+          "type": "string"
+        },
+        "target": {
+          "description": "Property name/path in request associated with error.",
+          "type": "string"
+        },
+        "details": {
+          "description": "Array with additional error details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudError"
+          }
+        }
+      },
+      "required": [
+        "code",
+        "message"
+      ]
+    },
+    "SubResource": {
+      "description": "Azure Synapse nested resource, which belongs to a workspace.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/AzureEntityResource"
+        }
+      ]
+    },
+    "LibraryListResponse": {
+      "type": "object",
+      "description": "Collection of libraries information",
+      "title": "Collection of libraries",
+      "properties": {
+        "nextLink": {
+          "type": "string",
+          "description": "The link to the next page of results, if any remaining results exist."
+        },
+        "value": {
+          "type": "array",
+          "description": "List of libraries",
+          "items": {
+            "$ref": "#/definitions/LibraryResource"
+          }
+        }
+      }
+    },
+    "LibraryResource": {
+      "type": "object",
+      "description": "A library resource",
+      "title": "Library artifact",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/LibraryResourceProperties",
+          "description": "Library/package properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "LibraryResourceProperties": {
+      "description": "Library/package properties",
+      "type": "object",
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Name of the library/package."
+        },
+        "path": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Location of library/package in storage account."
+        },
+        "containerName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Container name of the library/package."
+        },
+        "uploadedTimestamp": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The last update time of the library/package."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Type of the library/package."
+        },
+        "provisioningStatus": {
+          "readOnly": true,
+          "type": "string",
```

Comment:
>type [](start = 11, length = 4)

This should be modeled as an enum instead of a string.

Reply:
ProvisioningState in the other specs([link](https://github.com/Azure/azure-rest-api-specs/blob/71aa7928d23596ae43d69a8ece4beac3428da577/specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/bigDataPool.json#L398)) is also of type "string". that is the reason it is also of type string.